File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/PageDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder(builderMethodName = "pageBuilder")
@AllArgsConstructor
@NoArgsConstructor
public class PageDto {

  private int pageIndex;

  private int pageSize;

  private long hits;

  private int pagesTotal;
}

File: frontend/src/app/dto/page.dto.ts
export class PageDto {
  pageIndex = 0;

  pagesTotal = 1;

  pageSize = 10;
  hits = 0;


  constructor(pageIndex: number, pageSize: number, hits: number, pagesTotal: number) {
    this.pageIndex = pageIndex;
    this.pageSize = pageSize;
    this.hits = hits;
    this.pagesTotal = pagesTotal;
  }

  static initialPage(): PageDto {
    return new PageDto(0, 10, 0, 0);
  }
}
